galaxy/main/signals/handlers.py

```diff
@@ -44,35 +44,39 @@ def user_logged_in_handler(request, user, **kwargs):
     sanitized_username = username.lower().replace('-', '_')
 
     try:
-        namespace = models.ProviderNamespace.objects.get(name=username)
+        models.ProviderNamespace.objects.get(name=username)
         return
     except models.ProviderNamespace.DoesNotExist:
-        namespace = None
+        pass
 
-    # User is not associated with any Namespaces, so we'll attempt
-    # to create one, along with associated Provider Namespaces.
-    provider = models.Provider.objects.get(name__iexact="github")
+    # User is not associated with any Provider Namespaces, so we'll attempt
+    # to create one, along with associated Namespace.
 
     # if name doesn't exist, set it to login
     name = social.extra_data.get('name') or username
 
     defaults = {
```

Comment:
If you're reusing defaults dictionary, it's not the best approach to add query-specific params, then execute a query and then remove those specific params in order to execute another one.

Please use common dictionary and add query-specific params when they're required:


```
defaults = {
    ...
}

ns_defaults = {'name': sanitized_username, **defaults}
models.Namespace.objects.get_or_create(..., defaults=ns_defaults)